### PR TITLE
EXRLoader: adds support for ZIP compression

### LINKED ISSUE #17866: Feature Request: Add ZIP(S) compression support to EXRLoader
As suggested in https://github.com/mrdoob/three.js/pull/14895#issuecomment-421139939.

### PR DESCRIPTION
**Pull request by sciecode**
fixed #17866 

Introduces support for ZIP compression [EXR matcap example](https://rawcdn.githack.com/sciecode/three.js/4d6c275ef24759313f26af0d2a8af32d5e46701e/examples/webgl_loader_texture_exr.html)

~~I'm 90% sure it also adds support for ZIPS compression files, but I didn't manage to find any EXR file with that kind of compression. I would appreciate if someone could share one.~~ (see comment)

### CODE REVIEW

File: examples/js/loaders/EXRLoader.js [hunk 1]
```diff
@@ -1199,7 +1268,88 @@ THREE.EXRLoader.prototype = Object.assign( Object.create( THREE.DataTextureLoade
...
+
+						if ( EXRHeader.channels[ channelID ].pixelType === 1 ) { // half
+
+							for ( var x = 0; x < width; x ++ ) {
+
+								var idx = ( line_y * ( EXRHeader.channels.length * width ) ) + ( channelID * width ) + x;
+
+								switch ( this.type ) {
+
+									case FloatType:
```
Comment (WestLangley) [inline examples/js/loaders/EXRLoader.js]: add `THREE` namespace

File: examples/js/loaders/EXRLoader.js [hunk 2]
```diff
@@ -1199,7 +1268,88 @@ THREE.EXRLoader.prototype = Object.assign( Object.create( THREE.DataTextureLoade
...
+
+						} else if ( EXRHeader.channels[ channelID ].pixelType === 2 ) { // float
+
+							for ( var x = 0; x < width; x ++ ) {
+
+								var idx = ( line_y * ( EXRHeader.channels.length * width ) ) + ( channelID * width ) + x;
+
+								switch ( this.type ) {
+
+									case FloatType:
```
Comment (WestLangley) [inline examples/js/loaders/EXRLoader.js]: ditto

File: examples/js/loaders/EXRLoader.js (hunk 1 above)
Comment (sciecode) [reply examples/js/loaders/EXRLoader.js]: Good catch, thanks.

**Comment by sciecode** [comment]
Please, don't merge this just yet. I'm still studying a few unique cases.

**Comment by sciecode** [comment]
Managed to get my hands on a ZIPS file and fixed a few edge cases.
Working like a charm :100: :four_leaf_clover:

**Comment by WestLangley** [comment]
This looks good to me. Thanks!

**Comment by mrdoob** [comment]
Sweet!

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**